File: modules/animal-module/animal-module-rest/src/main/java/pl/adoptme/adopt/me/animal/AnimalDto.java
package pl.adoptme.adopt.me.animal;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import pl.adoptme.adopt.me.tags.TagDto;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@Builder

public class AnimalDto {
    private String id;
    private String name;
    private String age;
    private String attachmentId;
    private AnimalType type;
    private AnimalBreed breed;
    private AnimalColor color;
    private AnimalHair hair;
    private AnimalSize size;
    private Set<TagDto> tags;
    private List<HealthBooklet> healthBooklet;
    private String town;
    private String contact;
    private LocalDateTime createdOn;
    private String createdById;
    private LocalDateTime updateOn;
    private LocalDateTime deletedOn;
    private String deletedById;
}

File: adopt-me-front/src/api/animal/response/AnimalDto.ts
import {TagDto} from "../../tag/response/TagDto";
import {HealthBookletDto} from "./HealthBookletDto";
import {AnimalSizeEnum} from "../AnimalSizeEnum";
import {AnimalColorEnum} from "../AnimalColorEnum";
import {AnimalBreedEnum} from "../AnimalBreedEnum";
import {AnimalTypeEnum} from "../AnimalTypeEnum";
import {AnimalHairEnum} from "../AnimalHairEnum";

export interface AnimalDto{
    id:string
    name:string
    age:string
    attachmentId?: string
    tags:TagDto[]
    type:AnimalTypeEnum
    breed:AnimalBreedEnum
    color:AnimalColorEnum
    hair:AnimalHairEnum
    size:AnimalSizeEnum
    town:string
    contact:string
    healthBooklet?:HealthBookletDto[]
    createdOn: Date,
    createdById: string
    updatedOn: Date
    deletedOn?: Date,
    deletedById?: string
}